src/internal/types.ts

```diff
@@ -24,25 +33,54 @@ export type Mapped<T extends IterableContainer, K> = {
 export type IterableContainer<T = unknown> = ReadonlyArray<T> | readonly [];
 
 /**
- * We define a Simple record as one that doesn't define it's properties beyond
- * their general type. This is the case for string and number keys (and their
- * combination). Another way of thinking about simple records is that the number
- * of properties are unbound (they can be empty, or they can have 1000 props).
- *
- * TODO: Template string literals (e.g. `prefix_${number}`) should also be
- * considered simple, but we don't know how to check for them...
+ * Check if a type is a simple record. A simple record is one that has an
+ * unbounded number of properties. For example, a record with a union of string
+ * or number keys is simple, and so is a record with `prefix_${number}` keys.
  */
 export type IfSimpleRecord<
```

Comment:
Now that the difference is the boundness of the key this name isn't very good anymore, it doesn't communicate the difference well. 
```ts
Record<`${MyEnum}_suffix_${number}`, MyObject>
```

isn't any simpler than `{ a: number }`.

Reply:
renamed to `IfBoundedRecord` (and moved around the args)